### PR TITLE
Bump Rust version to 1.62 in build scripts.

### PR DESCRIPTION
**Pull request by abizjak**
## Purpose

Bump the base version in client to the latest main.

## Changes

No major changes other than version of the toolchains in build scripts.

The only code change is the removal of NonExistingRewardAccount variant which has been removed in base some time ago.

On the way, an obsolete Dockerfile has been removed. The file was clearly not used in a very long time (it uses stack resolver 16.20).

## Checklist

- [x] My code follows the style of this project.
- [x] The code compiles without warnings.
- [x] I have performed a self-review of the changes.
- [x] I have documented my code, in particular the intent of the
      hard-to-understand areas.
- [x] (If necessary) I have updated the CHANGELOG.

### CODE REVIEW

File: src/Concordium/Client/Output.hs [hunk 1]
```diff
@@ -719,11 +719,6 @@ showRejectReason verbose = \case
   Types.RejectedReceive{..} ->
     let (contractName, funcName) = Wasm.contractAndFunctionName receiveName
     in [i|'#{funcName}' in '#{contractName}' at #{showCompactPrettyJSON contractAddress} failed with code #{rejectReason}|]
-  Types.NonExistentRewardAccount a ->
```
Comment (EmilLa1) [inline src/Concordium/Client/Output.hs:722]: Just to be sure - this is obsolete (and always has been on mainnet) ?

**Review by EmilLa1 — APPROVED** [review]
Is the Dockerfile not used anymore? I can see it used an old GHC version compared to what we use now.

File: src/Concordium/Client/Output.hs (hunk 1 above)
Comment (abizjak) [reply src/Concordium/Client/Output.hs:722]: Correct. This was used a long time ago when a baker, when registering, would designate an account to be rewarded on.

This was removed at least for testnet 4 (which was the testnet around 6 months before mainnet).

**Comment by abizjak** [comment]
> Is the Dockerfile not used anymore? I can see it used an old GHC version compared to what we use now.

Yeah, it was clearly not used anymore since the build just does not work, and has not for a long time. There is no reason to build concordium-client in a docker image.

**Pull request merged**